distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2266,11 +2266,13 @@ async def test_handle_object_columns_p2p(c, s, a, b):
 
 
 @gen_cluster(client=True)
-async def test_reconcile_mismatching_partitions_p2p_shuffling(c, s, a, b):
+async def test_reconcile_partitions(c, s, a, b):
     def make_partition(i):
         """Return mismatched column types for every other partition"""
         if i % 2 == 1:
-            return pd.DataFrame({"a": np.random.random(10), "b": [True] * 10})
```

Comment:
There's a regression hidden here:

Previously, we could reconcile `bool` and `float` partitions. Arrow considers these as irreconcilable (personally I agree).

Reply:
Has this problem been reported as such or was this just a dummy unit test?

Reply:
IIRC, this was a dummy example to test reconciliation. I could easily see a scenario where someone applies a "boolean" UDF that fails to cast (some) numerical values to a `bool` though.